src/js/sharedEth/connectors.js

```diff
@@ -72,7 +72,44 @@ async function requiresAuthDialog(ethereum) {
 }
 
 async function connectWeb3(eth, ethereum, disallowAuthDialog = false, isAutoConnect = false) {
-  if (ethereum) {
+  if (ethereum && !ethereum.isTally ) {
+    let trxProvider = ethereum;
+
+    if (disallowAuthDialog && (await requiresAuthDialog(ethereum))) {
+      return {
+        networkId: null,
+        account: null,
+        ethereum: null,
+      };
+    }
+
+    //TODO: This is going to change in the future with EIP-1193
+    if (!isAutoConnect) {
+      ethereum.request({ method: 'eth_requestAccounts' });
+    }
+
+    setNewTrxProvider(eth, trxProvider);
+
+    let [account, _] = await getAccounts(eth);
+
+    let networkIdStr = await getNetworkId(eth);
+    let networkId = parseInt(networkIdStr);
+    if (networkId === NaN) {
+      networkId = null;
+    }
+
+    return { networkId, account, ethereum };
+  } else {
+    return {
+      networkId: null,
+      account: null,
+      ethereum: null,
+    };
+  }
+}
+
+async function connectTally(eth, ethereum, disallowAuthDialog = false, isAutoConnect = false) {
```

Comment:
This function looks identical to the normal connectWeb3() method. Can we instead just call the connectWeb3() function on line 91 in connectWalletPorts.js above?

Reply:
They are similar. The extra piece of logic is in the if statements on lines 75 and 112. Because of issues related to the opening of the correct wallet if Tally and MM are both installed this is needed to handle clicking of the separate buttons and create a better user experience

Reply:
let's extract the meat of the connect into a helper function then. Lines 76 - 101 above can be put into a helper function that both `connectWeb3` and `connectTally` can call.

Reply:
Good call. I have added a helper to remove redundant code